### PR TITLE
Progress on xCollideFast(1 func), xCutscene(2 funcs), xDecal(1 func and some header stuff) and xEntBoulder(4 funcs)

### PR DESCRIPTION
**Pull request by alexhdevries**

### CODE REVIEW

**Review by mattbruv — CHANGES_REQUESTED** [review]
Please remove commented out ASM pragmas as they are not used and could interfere with scraping files

File: src/Core/x/xCutscene.h [hunk 1]
```diff
@@ -92,11 +92,13 @@ struct xCutscene
 };
 
 extern uint32 gFrameCount;
+extern xCutscene sActiveCutscene;
```
Comment (mattbruv) [inline src/Core/x/xCutscene.h]: All variables should be defined/declared in their C++ files, not header files

Comment (alexhdevries) [reply src/Core/x/xCutscene.h]: So functions/structs in header files, variables in c++ files?

Comment (mattbruv) [reply src/Core/x/xCutscene.h]: Yes. Function declarations in header files, function implementations in C++ files, variable declarations in C++ files is the pattern we have stuck to  so far

**Review by mattbruv — APPROVED** [review]

**Pull request merged**